main.py

```diff
@@ -1,19 +1,18 @@
 
 
 import argparse
+import os.path
+import time
+
+import numpy
 from sklearn_extra.cluster import KMedoids
 import pandas as pd
 import sys
-DATABUFFER = 268435456 # size of quarter of a gigabyte
+import psutil
+import math
+DATABUFFER = 16*math.sqrt(psutil.virtual_memory().free) # 8 MB
+# import datetime
 
-# def decide_row_num_halt(path,cols=None):
-#     if path[-5:] == ".xlsx" or path[-4:] == ".xls":
-#         temp = pd.read_excel(path,usecols=cols,nrows=1)
-#     elif path[-4:] == ".csv":
-#         temp=pd.read_csv(path,usecols=cols,nrows=1)
-#     else:
-#         return
-#     row_num= DATABUFFER/ sys.getsizeof(temp)
 
 
 def load_data(path,startrows=None,endrows=None,cols=None):
```

Comment:
yeah.... No. 

Every function should be no more than 4 lines. And I mean it. EVERY single function. FOUR lines AT MOST (excl the def). NO EXCEPTIONS.

FAQ:
q: Why?
a: The long answer is because this is more readble, modular etc. The short answer: this is they way absolutely everyone does it and so there's really no point in trying to prove something else.

q: Ok I'll do it next time 
a: No, do it this time. Don't create tech debt. 

q: But what about this <insert whatever case>, you see because <insert reason>, this must be an exception!
a: NO EXCEPTIONS!!!!! FOUR LINES!!!!!!!!!

Reply:
Also, read this: https://github.com/jnguyen095/clean-code/blob/master/Clean.Code.A.Handbook.of.Agile.Software.Craftsmanship.pdf